cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -844,6 +861,21 @@ class TestAzureFileSystem : public ::testing::Test {
     AssertFileInfo(fs(), blob_path, FileType::NotFound);
   }
 
+  void TestDeleteDirSuccessHaveBlobWithTrailingSlash() {
```

Comment:
I would rename this to `TestNonEmptyDirWithTrailingSlash`. "have blob with trailing slash" can be misinterpreted as the blob path itself having a trailing slash.

Reply:
Done